packages/purpose-process/src/services/purposeService.ts

```diff
@@ -238,30 +238,38 @@ export function purposeServiceBuilder(
       logger.info(`Retrieving Purpose ${purposeId}`);
 
       const purpose = await retrievePurpose(purposeId, readModelService);
-      const eservice = await retrieveEService(
-        purpose.data.eserviceId,
-        readModelService
-      );
 
-      const tenantKind = await retrieveTenantKind(
-        organizationId,
-        readModelService
-      );
+      const [eservice, activeConsumerDelegation, tenantKind] =
+        await Promise.all([
+          retrieveEService(purpose.data.eserviceId, readModelService),
+          readModelService.getActiveConsumerDelegationByPurpose({
+            eserviceId: purpose.data.eserviceId,
+            consumerId: purpose.data.consumerId,
+          }),
+          retrieveTenantKind(organizationId, readModelService),
+        ]);
+
+      const delegationId = activeConsumerDelegation
+        ? activeConsumerDelegation.id
+        : undefined;
+      const basePurpose = { ...purpose.data, delegationId };
```

Comment:
Why this change?
In theory, the delegationId should already be present inside the retrieved purpose when a delegator created the purpose.